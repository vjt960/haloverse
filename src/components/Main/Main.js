import React from 'react';
import { connect } from 'react-redux';
import { Showcase } from '../../containers/Showcase/Showcase';
import { Placeholder } from '../Placeholder/Placeholder';
import Form from '../../containers/Form/Form';
import Profile from '../../containers/Profile/Profile';

export const Main = props => {
  return (
    <main>
      {props.maps.length && <Showcase maps={props.maps} />}
      <Form />
      {props.stats ? <Profile /> : <Placeholder />}
    </main>
  );
};

export const mapStateToProps = state => ({
  maps: state.maps,
  stats: state.stats
});

export default connect(mapStateToProps)(Main);
