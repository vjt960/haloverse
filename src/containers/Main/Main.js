import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import React from 'react';
import { Loading } from '../../components/Loading/Loading';
import { Placeholder } from '../../components/Placeholder/Placeholder';
import { Showcase } from '../Showcase/Showcase';
import Form from '../Form/Form';
import Profile from '../Profile/Profile';

export const Main = props => {
  return (
    <main>
      {props.maps.length ? (
        <Showcase maps={props.maps} />
      ) : (
        <Loading target="showcase" />
      )}
      <Form />
      {!props.stats && <Placeholder />}
      {props.stats && <Profile />}
    </main>
  );
};

export const mapStateToProps = state => ({
  maps: state.maps,
  stats: state.stats,
  userIsLoading: state.userIsLoading
});

Main.propTypes = {
  maps: PropTypes.array.isRequired,
  stats: PropTypes.object.isRequired,
  userIsLoading: PropTypes.bool
};

export default connect(mapStateToProps)(Main);
