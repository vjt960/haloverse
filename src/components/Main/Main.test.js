import React from 'react';
import { shallow, mount } from 'enzyme';
import { Main, mapStateToProps } from './Main';

describe('Main', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<Main maps={[1, 2, 3]} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot when conditionally rendering Showcase', () => {
    const altWrapper = shallow(<Main maps={{ length: false }} />);
    expect(altWrapper).toMatchSnapshot();
  });

  it('should match snapshot when conditionally rendering Placeholder', () => {
    const altWrapper = shallow(<Main maps={{ length: false }} stats={true} />);
    expect(altWrapper).toMatchSnapshot();
  });
});

describe('mapStateToProps', () => {
  it('should return an array of map objects', () => {
    const mockState = {
      maps: ['map1', 'map2', 'map3']
    };

    const expected = {
      maps: ['map1', 'map2', 'map3']
    };
    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps).toEqual(expected);
  });

  it('should return an object containing user stats', () => {
    const mockState = {
      stats: { one: 'one', two: 'two' }
    };

    const expected = {
      stats: { one: 'one', two: 'two' }
    };
    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps).toEqual(expected);
  });
});
