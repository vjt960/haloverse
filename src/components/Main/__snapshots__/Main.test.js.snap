// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should match snapshot 1`] = `ShallowWrapper {}`;
